### DISCUSSION TITLE
Where can I find it, please #include <libredwg/libredwg.h>

### DISCUSSION THREAD

**Discussion by iisaxin** [LibreDWG/libredwg#921] [Q&A]
Hi, this is a program I found on the Internet. It can read the actual name of the anonymous block, but I don't have that library. Where can I find it? Thank you all for your help.
 Here are the programs I found:

#include <stdio.h>
**#include <libredwg/libredwg.h>**

void print_block_names(DWG_BlockRecordHandle blockRecord) {
    DWG_BlockHandle block = dwg_block_open(blockRecord);
    if (block != NULL) {
        **const char *blockName = dwg_block_name(block);
        const char *blockRealName = dwg_block_real_name(block);**

        printf("Block Name: %s\n", blockName);
        printf("Block Real Name: %s\n", blockRealName);

        dwg_block_close(block);
    }
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Usage: %s <input.dwg>\n", argv[0]);
        return 1;
    }

    const char *inputFileName = argv[1];

    // Initialize the LibreDWG library
    dwglib_initialize();

    // Open the DWG file for reading
    DWG_FileHandle dwgFile = dwg_open_file(inputFileName, DWG_READ);
    if (dwgFile == NULL) {
        printf("Failed to open DWG file: %s\n", inputFileName);
        dwglib_finish();
        return 1;
    }

    // Read the DWG file header
    DWG_Header header;
    if (dwg_get_header(dwgFile, &header) != DWG_SUCCESS) {
        printf("Failed to read DWG file header.\n");
        dwg_close_file(dwgFile);
        dwglib_finish();
        return 1;
    }

    // Check if the DWG file has block records
    if (header.flags & DWG_HEADER_FLAG_BLOCK_RECORDS) {
        printf("DWG file has block records.\n");

        // Read the block records
        DWG_BlockRecordIteratorHandle blockRecordIter = dwg_block_record_iterator(dwgFile);
        if (blockRecordIter != NULL) {
            while (dwg_block_record_iterator_next(blockRecordIter)) {
                DWG_BlockRecordHandle blockRecord = dwg_block_record_iterator_get(blockRecordIter);
                print_block_names(blockRecord);
            }

            dwg_block_record_iterator_close(blockRecordIter);
        }
    } else {
        printf("DWG file does not have block records.\n");
    }

    // Close the DWG file and finish
    dwg_close_file(dwgFile);
    dwglib_finish();

    return 0;
}

**Comment by rurban** [comment]
Sorry, that's certainly not us. Someone else is abusing our name. Look for some dwg library with dwglib_initialize().
maybe it's on top of our library.

**Reply by iisaxin** [reply]
Thank you for your patient answer, in fact, I also encountered the problem to find the answer on the network, I encountered the problem is to read the anonymous block, read back the name is * U, but the actual name did not get, so I am very troubled, can you help me to answer this question?
